File: common/src/main/java/com/ssblur/scriptor/helpers/targetable/EntityTargetable.java
package com.ssblur.scriptor.helpers.targetable;

import net.minecraft.world.entity.Entity;

public class EntityTargetable extends Targetable {
  Entity targetEntity;

  public EntityTargetable(Entity entity) {
    super(entity.level, entity.position());
    targetEntity = entity;
  }

  public Entity getTargetEntity() {
    return targetEntity;
  }
}

File: common/src/main/kotlin/com/ssblur/scriptor/helpers/targetable/EntityTargetable.kt
package com.ssblur.scriptor.helpers.targetable

import net.minecraft.world.entity.Entity

open class EntityTargetable(var targetEntity: Entity): Targetable(targetEntity.level(), targetEntity.position()) {
    override val entityYRotation: Float? = if (targetEntity.getYRot() > 0) {
        targetEntity.getYRot()
    } else {
        targetEntity.getYRot() + 360
    }

    override val entityCoarseYRotation: Float? = when (entityYRotation!!.toInt()) {
        in 0..45 -> 0f // South
        in 315..360 -> 0f // South
        in 45..135 -> 90f // East
        in 135..225 -> 180f // North
        else -> 270f // West
    }
}
